### PR TITLE
👌 IMPROVE: Second part of workchain module

### PR DESCRIPTION
**Pull request by mbercx**
Make several small changes throughout the second part of the
"Workflows - workchain" module.

* Linguistical and format improvements.
* Add exercises and solutions
* Turn "Adding complexity" section into exercise

Also rename the directory that stores the `.py` code snippets to correspond to
the name of the work chain markdown file.

### CODE REVIEW

**Comment by mbercx** [comment]
@flavianojs this doesn't contain the changes to the final section on including calculation jobs yet, will add those in a separate PR to make reviewing easier.

File: docs/sections/workflows/workchain.md [hunk 1]
```diff
@@ -380,120 +418,198 @@ That is because it is a new data node that was created by the calculation functi
...
+Based on the concepts you've learned so far, add an extra multiplication step by doing the following:
 
-Increasing the level of complexity, we want to write a work chain which receives three inputs, multiply the first two inputs and add to that result the value of the third input.
-Also, we want the work chain to output both the product of the first two inputs and the final result.
-Keep in mind that in real-life examples, in the steps where we are doing mere arithmetic, one shoudl expect complex tasks such as running an external code.
-Our code may seem like this:
+* Rename the work chain to `MultiplyAddWorkChain`, since we'll be adding an extra multiplication step.
+* Write a calculation function called `multiplication`, that takes two `Int` nodes and returns their product.
+* Add a new `Int` input to the `MultiplyAddWorkChain` `spec`, labeled `'z'`.
+* Add a new step to the _outline_ of the work chain called `multipy`.
```
Comment (flavianojs) [inline docs/sections/workflows/workchain.md]:
```suggestion
* Add a new step to the _outline_ of the work chain called `multipy`.
  Make it the first step of the _outline_.
```

File: docs/sections/workflows/workchain.md [hunk 2]
```diff
@@ -380,120 +418,198 @@ That is because it is a new data node that was created by the calculation functi
...
+  When defining the method, use the `multiplication` calculation function to multiply the `x` and `y` inputs.
+  Then pass the results to the `add` step using the context.
+* In the `add()` method, sum the result of the multiplication with the third input `z`.
```
Comment (flavianojs) [inline docs/sections/workflows/workchain.md]:
```suggestion
* In the `add()` method, sum the result of the multiplication with the third input `z`.
  Pass the result to the context.
```

File: docs/sections/workflows/workchain.md [hunk 3]
```diff
@@ -380,120 +418,198 @@ That is because it is a new data node that was created by the calculation functi
...
+* In the `result()` method, also attach the result of the multiplication as an output.
+  You can use `product` as the label for the output link, for example.
```
Comment (flavianojs) [inline docs/sections/workflows/workchain.md]:
```suggestion
* In the `result()` method, output the result of step `add()`.
  Also, attach the result of the multiplication as an output (you need to declare another output for this in the `define` method).
  You can use `product` as the label for the output link, for example.
```

**Review by flavianojs — APPROVED** [review]
Hi Marnik, it looks very nice. I have only a few suggestions for the last exercise.
In any case, I am already approving this PR for merging.

**Comment by mbercx** [comment]
Thanks for the review, @flavianojs! 👌

**Pull request merged**